Tavisca.Bootcamp.LanguageBasics.Exercise2/Program.cs

```diff
@@ -13,18 +13,58 @@ static void Main(string[] args)
             Console.ReadKey(true);
         }
 
-        private static void Test(string[] postTimes, string[] showTimes, string expected)
-        {
+        private static void Test(string[] postTimes, string[] showTimes, string expected)
+        {
             var result = GetCurrentTime(postTimes, showTimes).Equals(expected) ? "PASS" : "FAIL";
             var postTimesCsv = string.Join(", ", postTimes);
             var showTimesCsv = string.Join(", ", showTimes);
-            Console.WriteLine($"[{postTimesCsv}], [{showTimesCsv}] => {result}");
+            Console.WriteLine($"[{postTimesCsv}], [{showTimesCsv}] => {result}");
         }
 
-        public static string GetCurrentTime(string[] exactPostTime, string[] showPostTime)
-        {
+        public static string GetCurrentTime(string[] exactPostTime, string[] showPostTime)
+        {
             // Add your code here.
-            throw new NotImplementedException();
+            //check for self-contradiction
+            for(int i=0; i<exactPostTime.Length-1; i++)
+            {
+                for(int j=i+1; j<exactPostTime.Length; j++)
+                {
+                    if(exactPostTime[i].Equals(exactPostTime[j]))
+                    {if(!(showPostTime[i].Equals(showPostTime[j])))
+                        {
+                            return "impossible";
+                        }
+                    }
+                }
+			}
+			
+            string resulttime="";
+
+            for(int i = 0; i<exactPostTime.Length; i++) 
+            {
+                DateTime PostedDateTime, CurrentDateTime;
+            	PostedDateTime = DateTime.Parse(exactPostTime[i]);
+                CurrentDateTime = PostedDateTime;
+
+                // cases for second, minute and hour
+                if(showPostTime[i].Contains("seconds"))
+                {
+                    CurrentDateTime = PostedDateTime;
+                }
+                else if(showPostTime[i].Contains("minutes"))
+                { 
+                  CurrentDateTime = PostedDateTime.AddMinutes(Convert.ToDouble(showPostTime[i].Substring(0,2)));

```

Comment:
Instead of trying to cram the entire implementation into a single line, it would be more useful to break up the bits into refactorable values. Eg. `Convert.ToDouble(showPostTime[i].Substring(0,2))` could be moved to a `timeToAdjust` variable which can be set conditionally and then simply added to the result in the end. It would also make the code more readable. 
Incidentally a comment declaring your approach would be quite helpful. Something like 
```
// Our approach for this problem is to identify the time interval that we need to adjust the postedDateTime with to 
// arrive at the current time. We can identify this interval based on the showPostTime options.